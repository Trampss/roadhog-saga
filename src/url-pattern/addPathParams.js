import { encodeParams, isEmpty } from './utils'

export default url => (pathParams) => {
  if (isEmpty(pathParams)) return url
  return `${url}${url.endsWith('/') ? '' : '/'}${encodeParams(pathParams).join('/')}`
}
