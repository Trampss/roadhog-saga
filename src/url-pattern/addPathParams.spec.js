/* eslint-env jest */
import addPathParams from './addPathParams'

describe('addPathParams', () => {
  it('should return url without modification', () => {
    expect(addPathParams('http://an-url')()).toMatchSnapshot()
  })

  it('should return url with path params (no trailing /)', () => {
    expect(addPathParams('http://an-url')(['some', 'param'])).toMatchSnapshot()
  })

  it('should return url with path params (trailing /)', () => {
    expect(addPathParams('http://an-url/')(['some', 'param'])).toMatchSnapshot()
  })
})
