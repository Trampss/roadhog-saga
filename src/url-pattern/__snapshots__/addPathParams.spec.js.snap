exports[`addPathParams should return url with path params (no trailing /) 1`] = `"http://an-url/some/param"`;

exports[`addPathParams should return url with path params (trailing /) 1`] = `"http://an-url/some/param"`;

exports[`addPathParams should return url without modification 1`] = `"http://an-url"`;
